### PR TITLE
Hotfix/1.0.2

### PR DESCRIPTION
**Pull request by turegjorup**

### CODE REVIEW

File: ding_pretix.module [hunk 1]
```diff
@@ -261,13 +261,34 @@ function ding_pretix_form_ding_event_node_form_alter(&$form, &$form_state, $form
   // Service settings.
   $service_settings = variable_get('ding_pretix', []);
 
+  // If we are cloning we need to find and set the pretix settings from the event being cloned from
```
Comment (cableman) [inline ding_pretix.module]: All comments should be a sentence with full stop.

File: ding_pretix.module [hunk 2]
```diff
@@ -316,16 +337,21 @@ function ding_pretix_form_ding_event_node_form_alter(&$form, &$form_state, $form
       $options[$element['psp_value']] = $element['psp_name'];
     }
 
-    // PSP is a code for accounting. Once set, we don't allow this to be
+    // PSP is a code for accounting. If an event has orders, we don't allow this to be
     // changed, as this would invalidate the accounts.
-    $disabled = empty($pretix_node_info['psp_element']) ? FALSE : TRUE;
-    $description = $disabled ? t('For accounting reasons the PSP element cannot be changed.') : t('Select the PSP element the ticket sales should be registered under.');
+    if($pretix_node_info['pretix_slug']) {
```
Comment (cableman) [inline ding_pretix.module]: Missing space "if()".

File: ding_pretix.module [hunk 3]
```diff
@@ -407,3 +433,28 @@ function ding_pretix_node_presave($node) {
...
+/**
+ * Implements hook_node_clone_alter().
+ */
+function ding_pretix_clone_node_alter(&$node, $context) {
+  // We only handles event nodes.
+  if ($node->type === 'ding_event') {
+    module_load_include('inc', 'ding_pretix', 'includes/ding_pretix.api_module');
+    $service_settings = variable_get('ding_pretix', []);
+
+    $node_settings = $node->clone_from_original_nid ? _ding_pretix_get_pretix_node_info($node->clone_from_original_nid) : null;
```
Comment (cableman) [inline ding_pretix.module]: null should be NULL

File: ding_pretix.module [hunk 4]
```diff
@@ -407,3 +433,28 @@ function ding_pretix_node_presave($node) {
...
+
+    $node->pretix = [
+      'nid' => $node->nid,
+      'capacity' => (int) $node_settings['capacity'],
+      'maintain_copy' => $node_settings['maintain_copy'],
+      'psp_element' => $node_settings['psp_element'],
+      'ticket_form' => $node_settings['ticket_form'],
+    ];
+
+    if($node_settings['maintain_copy']) {
```
Comment (cableman) [inline ding_pretix.module]: Missing space "if(".

File: includes/ding_pretix.api_module.inc [hunk 5]
```diff
@@ -223,6 +223,30 @@ function _ding_pretix_get_quota($items, $service_settings, $pretix_slug) {
   return $quota ?? NULL;
 }
 
+/**
+ * Check if there are orders for the given event in Pretix.
+ *
+ * @param $service_settings
```
Comment (cableman) [inline includes/ding_pretix.api_module.inc]: Missing parameter documentation.

File: includes/ding_pretix.api_module.inc [hunk 6]
```diff
@@ -223,6 +223,30 @@ function _ding_pretix_get_quota($items, $service_settings, $pretix_slug) {
...
+ * @param $pretix_slug
```
Comment (cableman) [inline includes/ding_pretix.api_module.inc]: Missing parameter documentation.

File: includes/ding_pretix.api_module.inc [hunk 7]
```diff
@@ -223,6 +223,30 @@ function _ding_pretix_get_quota($items, $service_settings, $pretix_slug) {
...
+ *
+ * @return mixed
```
Comment (cableman) [inline includes/ding_pretix.api_module.inc]: Missing return documentation.

File: includes/ding_pretix.api_module.inc [hunk 8]
```diff
@@ -223,6 +223,30 @@ function _ding_pretix_get_quota($items, $service_settings, $pretix_slug) {
...
+ */
+function _ding_pretix_has_orders($service_settings, $pretix_slug) {
+  $options = [
```
Comment (cableman) [inline includes/ding_pretix.api_module.inc]: Could at static cache be used here?

**Review by cableman — CHANGES_REQUESTED** [review]

**Comment by turegjorup** [comment]
@cableman PR updated

File: ding_pretix.module [hunk 9]
```diff
@@ -73,6 +73,11 @@ function ding_pretix_field_formatter_info() {
 
 /**
  * Implements hook_node_insert().
+ *
+ * @param $node
+ *   The node that is being created.
+ *
+ * @throws Exception if one of the INSERTs failed
```
Comment (cableman) [inline ding_pretix.module]: The exception type should be defined if possible and comment on next line. See https://www.drupal.org/docs/develop/coding-standards/api-documentation-and-comment-standards#throws

Comment (cableman) [reply ding_pretix.module]: This is true for all @throws in the code

File: ding_pretix.module [hunk 10]
```diff
@@ -193,6 +203,9 @@ function ding_pretix_node_update($node) {
 
 /**
  * Implements hook_node_delete().
+ *
+ * @param $node
+ *   The node that is being deleted.
```
Comment (cableman) [inline ding_pretix.module]: Hooks used short form comments normally: https://www.drupal.org/docs/develop/coding-standards/api-documentation-and-comment-standards#hookimpl

Comment (cableman) [reply ding_pretix.module]: Please don't kill the messenger....

File: ding_pretix.module [hunk 11]
```diff
@@ -283,19 +334,19 @@ function ding_pretix_form_ding_event_node_form_alter(&$form, &$form_state, $form
   ];
 
   // We don't allow updates to price after the event is created in pretix -
-  // must be updated in there.
-  if ($pretix_node_info['pretix_slug']) {
+  // must be updated in pretix.
```
Comment (cableman) [inline ding_pretix.module]: Comment should be an sentence stating with capital letter and end with .

File: ding_pretix.module [hunk 12]
```diff
@@ -283,19 +334,19 @@ function ding_pretix_form_ding_event_node_form_alter(&$form, &$form_state, $form
...
   // We don't allow manual change of the ticket link if pretix is used.
-  if ($pretix_node_info['pretix_slug']) {
+  if ($pretix_node_info['maintain_copy']) {
     $form['field_ding_event_ticket_link']['#disabled'] = TRUE;
     $form['field_ding_event_ticket_link']['und'][0]['#description'] = t('This field is managed by pretix for this event.');
   }
 
   // We don't allow updates to capacity after the event is created in pretix -
-  // must be updated in there.
+  // must be updated in pretix.
```
Comment (cableman) [inline ding_pretix.module]: Same here with comment

File: includes/ding_pretix.api_module.inc [hunk 13]
```diff
@@ -126,12 +215,35 @@ function _ding_pretix_update_live($is_live, $service_settings, $pretix_slug) {
 /**
  * Enable or disable the 'PDF ticket plugin' in pretix.
  *
- * @param $data
- * @param $use_pdf_tickets
- * @param $service_settings
- * @param $pretix_slug
+ * @param string $data
+ * @param bool $use_pdf_tickets
```
Comment (cableman) [inline includes/ding_pretix.api_module.inc]: Documentation messing for $data and $use_pdf_tickets.

File: includes/ding_pretix.api_module.inc [hunk 14]
```diff
@@ -164,11 +276,34 @@ function _ding_pretix_update_pdf_plugin($data, $use_pdf_tickets, $service_settin
  * Update the size of the given quota.
  *
  * @param $quota
```
Comment (cableman) [inline includes/ding_pretix.api_module.inc]: Missing documentation for $quota.

File: includes/ding_pretix.api_module.inc [hunk 15]
```diff
@@ -33,10 +33,32 @@ function ding_pretix_get_shop_url($nid) {
...
+ *     status code.
+ *   - redirect_url: If redirected, a string containing the URL of the redirect
+ *     target.
+ *   - error: If an error occurred, the error message. Otherwise not set.
+ *   - headers: An array containing the response headers as name/value pairs.
+ *     HTTP header names are case-insensitive (RFC 2616, section 4.2), so for
+ *     easy access the array keys are returned in lower case.
+ *   - data: A string containing the response body that was received.
  *
  * @throws Exception
```
Comment (cableman) [inline includes/ding_pretix.api_module.inc]: Missing documentation?

File: includes/ding_pretix.api_module.inc [hunk 16]
```diff
@@ -60,11 +82,34 @@ function _ding_pretix_create_event($node, $service_settings) {
...
```
Comment (cableman) [inline includes/ding_pretix.api_module.inc]: Missing documentation?

**Review by cableman — CHANGES_REQUESTED** [review]

**Review by cableman — APPROVED** [review]

**Pull request merged**